src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -907,6 +907,65 @@ public override IOperation VisitWhileLoop(IWhileLoopOperation operation, int? ca
             return null;
         }
 
+        public override IOperation VisitVariableDeclarationGroup(IVariableDeclarationGroupOperation operation, int? captureIdForResult)
+        {
+            // We erase variable declarations from the control flow graph, as variable lifetime information is contained in a parallel data structure.
+            foreach (var declaration in operation.Declarations)
+            {
+                VisitVariableDeclaration(declaration, captureIdForResult);
+            }
+
+            return null;
+        }
+
+        public override IOperation VisitVariableDeclaration(IVariableDeclarationOperation operation, int? captureIdForResult)
+        {
+            foreach (var declarator in operation.Declarators)
+            {
+                ILocalSymbol localSymbol = declarator.Symbol;
+
+                IOperation initializer = null;
+                SyntaxNode assignmentSyntax = null;
+                if (declarator.Initializer != null)
+                {
+                    initializer = Visit(declarator.Initializer);
+                    assignmentSyntax = declarator.Syntax;
+                }
+
+                if (operation.Initializer != null)
+                {
+                    IOperation operationInitializer = Visit(operation.Initializer);
+                    assignmentSyntax = operation.Syntax;
+                    if (initializer != null)
+                    {
+                        initializer = new InvalidOperation(ImmutableArray.Create(initializer, operationInitializer), semanticModel: null, operation.Syntax, type: localSymbol.Type, constantValue: default, isImplicit: true);
+                    }
+                    else
+                    {
+                        initializer = operationInitializer;
+                    }
+                }
+
+                if (initializer != null)
+                {
+                    // We can't use the IdentifierToken as the syntax for the local reference, so we use the entire declarator as the node
+                    var localRef = new LocalReferenceExpression(localSymbol, isDeclaration: true, semanticModel: null, declarator.Syntax, localSymbol.Type, constantValue: default, isImplicit: true);
+
+                    var statement = new ExpressionStatement(new SimpleAssignmentExpression(localRef, isRef: localSymbol.IsRef, initializer, semanticModel: null, assignmentSyntax, localRef.Type, constantValue: default, isImplicit: true),
```

Comment:
>new ExpressionStatement [](start = 36, length = 23)

I think we can skip creation of an ExpressionStatement here. It is fine to use simple assignment as a statement.

Reply:
I considered this, but my thought was that for analyzer authors it would be helpful if:
```C#
int a = 1;
int b;
b = 1;
```
ended up with the same shape in the flow graph, namely an IExpressionStatementOperation wrapping an ISimpleAssignmentOperation.

---
In reply to: [166782118](https://github.com/dotnet/roslyn/pull/24692#discussion_r166782118) [](ancestors = 166782118)

Reply:
With IOperation from syntax our hand was forced by syntax. There is no constraint like that for synthetic nodes, and I think we should keep amount synthesized nodes to a necessary minimum

---
In reply to: [166783230](https://github.com/dotnet/roslyn/pull/24692#discussion_r166783230) [](ancestors = 166783230,166782118)

Reply:
I'm going to leave this as is for now. During design discussion next week, we'll talk about removing all ExpressionStatements from the Control Flow graph, as they don't really serve a purpose.

---
In reply to: [166784473](https://github.com/dotnet/roslyn/pull/24692#discussion_r166784473) [](ancestors = 166784473,166783230,166782118)